snowfall/models/transformer.py

```diff
@@ -585,23 +585,10 @@ def encoder_padding_mask(supervisions: Optional[Dict] = None) -> Optional[Tensor
          supervisions['start_frame'],
          supervisions['num_frames']), 1).to(torch.int32)
 
-    indices = torch.argsort(supervision_segments[:, 1], descending=False)
-    supervision_segments = supervision_segments[indices]
-
-    indices = torch.argsort(supervision_segments[:, 0], descending=False)
-    supervision_segments = supervision_segments[indices]
-
-    previous_idx = -1
-    cur_length = 0
-    lengths = []
+    lengths = [0 for _ in range(int(max(supervision_segments[:, 0])) + 1)]
```

Comment:
Is there a reason not to make it an empty list and append each item in the loop below?

Reply:
If make a empty list and append in the loop, we have to decide whether it is the first sentence of this 'sequence_idx'. If true, we should append it. if not, we should update the last value in the list. I think initialize them first could keep the loop simple. But the other way is also good to me.

Reply:
OK got it - I didn't notice that `sequence_idx` doesn't have to be sorted ascendingly.